src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,178 @@
+
+const thoughtViewLookUp = (value, context, rank, cursor, contextChain, ignoreChildren) => {
+  const { showHiddenThoughts } = store.getState()
+
+  if (contextChain.length > 1 && head(contextChain).length === 1) {
+    return
+  }
+  const derivedCursor = !ignoreChildren && contextChain.length > 1 ? getDerivedCursor(contextChain) : cursor
+  const derivedContextRanked = contextOf(derivedCursor)
+  // map is expensive than duplicate condition check hence using the former
+  const derivedContext = !ignoreChildren && contextChain.length > 1 ? contextOf(derivedCursor).map(c => c.value) : context
+
+  const firstChild = firstChildOfThought(derivedCursor || RANKED_ROOT, showHiddenThoughts)
+  if (!ignoreChildren && firstChild) {
+    return { nextThoughts: unroot(cursor.concat(firstChild)), contextChain: [] }
+  }
+  const sortPreference = getSortPreference(meta(pathToContext(derivedContext)))
+  const siblings = sortPreference === 'Alphabetical' ? getThoughtsSorted(derivedContext) : getThoughtsRanked(derivedContext)
+
+  const contextHead = ((contextChain).length > 1 && head(contextChain).length < 3) ? head(contextChain)[0].value : head(derivedContext)
+  if (siblings.length === 0) {
+    return getNextThoughtsWithContextChain(contextHead, contextOf(derivedContext), rank, contextOf(derivedCursor), derivedContextRanked, contextChain.length > 1 ? [contextChain.slice(-1)] : contextChain)
+  }
+
+  const nextSibling = thoughtNextSibling(value, derivedContext, rank, showHiddenThoughts)
+  if (nextSibling) {
+    return { nextThoughts: unroot(derivedContextRanked.concat(nextSibling)), contextChain: contextOf(contextChain) }
+  }
+  // look for uncle/ancestor sibling
+  else {
+    const parentThought = derivedCursor.slice(-2, -1)[0]
+    const parentContext = context.length === 1 ? [ROOT_TOKEN] : contextOf(derivedContext)
+    const contextChainForParentThought = [...contextOf(contextChain), contextOf(head(contextChain))]
+    const parentCursor = contextOf(derivedCursor)
+
+    const cursorToFirstThoughtInContext = contextOf(contextChain.length > 1 ? contextChain.flat() : derivedCursor)
+    // restricts from working with multilevel context chains
+    const rankedContextOfCurrentContext = contextOf(contextChain)[0]
+    const contextChainTillFirstChildOfContext = [...contextOf(contextChain), [head(contextChain)[0]]]
+    const firstThoughtInContext = head(contextChain)[0]
+
+    // reached last child of last thought
+    if (!parentThought) {
+      return { nextThoughts: [siblings[0]] || [ROOT_TOKEN], contextChain: [] }
+    }
+    const thoughtsWithContextChain = thoughtViewLookUp(parentThought.value, parentContext, parentThought.rank, parentCursor, contextChainForParentThought, true)
+    return thoughtsWithContextChain || contextViewLookup(firstThoughtInContext.value, rankedContextOfCurrentContext.map(c => c.value), firstThoughtInContext.rank, cursorToFirstThoughtInContext, rankedContextOfCurrentContext, contextChainTillFirstChildOfContext, true)
```

Comment:
Use `pathToContext` instead of `.map(c => c.value)`.

Reply:
Updated